product-scenarios/scenarios-commons/src/main/java/org/wso2/identity/scenarios/commons/SAML2SSOTestBase.java

```diff
@@ -0,0 +1,891 @@
+        }
+        post.setEntity(new UrlEncodedFormEntity(urlParameters));
+
+        return client.execute(post);
+    }
+
+    /**
+     * Build SAML2 Authentication Request
+     *
+     * @param spEntityID
+     * @param nameIDFormat
+     * @param isPassiveAuthn
+     * @param isForceAuthn
+     * @param httpBinding
+     * @param acsUrl
+     * @param destinationUrl
+     * @return
+     */
+    protected AuthnRequest buildAuthnRequest(String spEntityID, String nameIDFormat, boolean isPassiveAuthn,
+                                             boolean isForceAuthn, String httpBinding, String acsUrl,
+                                             String destinationUrl) {
+        IssuerBuilder issuerBuilder = new IssuerBuilder();
+        Issuer issuer = issuerBuilder.buildObject(SAML_ASSERTION_URN, ISSUER, NAMESPACE_PREFIX);
+        issuer.setValue(spEntityID);
+
+        NameIDPolicyBuilder nameIdPolicyBuilder = new NameIDPolicyBuilder();
+        NameIDPolicy nameIdPolicy = nameIdPolicyBuilder.buildObject();
+        nameIdPolicy.setFormat(nameIDFormat);
+        nameIdPolicy.setSPNameQualifier(ISSUER);
+        nameIdPolicy.setAllowCreate(true);
+
+        AuthnContextClassRefBuilder authnContextClassRefBuilder = new AuthnContextClassRefBuilder();
+        AuthnContextClassRef authnContextClassRef =
+                authnContextClassRefBuilder.buildObject(SAML_ASSERTION_URN, AUTHN_CONTEXT_CLASS_REF,
+                        NAMESPACE_PREFIX);
+        authnContextClassRef.setAuthnContextClassRef(PASSWORD_PROTECTED_TRANSPORT_CLASS);
+
+
+        RequestedAuthnContextBuilder requestedAuthnContextBuilder =
+                new RequestedAuthnContextBuilder();
+        RequestedAuthnContext requestedAuthnContext = requestedAuthnContextBuilder.buildObject();
+        requestedAuthnContext.setComparison(AuthnContextComparisonTypeEnumeration.EXACT);
+        requestedAuthnContext.getAuthnContextClassRefs().add(authnContextClassRef);
+
+        DateTime issueInstant = new DateTime();
+
+        AuthnRequestBuilder authRequestBuilder = new AuthnRequestBuilder();
+        AuthnRequest authRequest =
+                authRequestBuilder.buildObject(SAML_PROTOCOL_URN, AUTHN_REQUEST, NAMESPACE_PREFIX);
+
+        authRequest.setForceAuthn(isForceAuthn);
+        authRequest.setIsPassive(isPassiveAuthn);
+        authRequest.setIssueInstant(issueInstant);
+        authRequest.setProtocolBinding(httpBinding);
+        authRequest.setAssertionConsumerServiceURL(acsUrl);
+        authRequest.setIssuer(issuer);
+        authRequest.setNameIDPolicy(nameIdPolicy);
+        authRequest.setRequestedAuthnContext(requestedAuthnContext);
+        authRequest.setID(createID());
+        authRequest.setVersion(SAMLVersion.VERSION_20);
+        authRequest.setDestination(destinationUrl);
+
+        return authRequest;
+    }
+
+    protected AuthnRequest buildAuthnRequest(SAMLSSOServiceProviderDTO samlssoSPDTO, boolean isPassiveAuthn,
```

Comment:
add method descriptions, since these can be heavily used by derived child classes.